src/components/contextual/stake/StakePreview.vue

```diff
@@ -33,35 +35,63 @@ const { balanceFor, getToken } = useTokens();
 const { fNum2 } = useNumbers();
 const { t } = useI18n();
 const queryClient = useQueryClient();
-const { stakeBPT, getGaugeAddress } = useStaking(props.pool);
+const { stakeBPT, unstakeBPT, getGaugeAddress, getStakedShares } = useStaking(
+  props.pool.address
+);
 const { getTokenApprovalActionsForSpender } = useTokenApprovalActions(
   [props.pool.address],
   ref([balanceFor(props.pool.address).toString()])
 );
 
+const stakeAction = {
+  label: t('stake'),
+  loadingLabel: t('staking.staking'),
+  confirmingLabel: t('confirming'),
+  action: stakeBPT,
+  stepTooltip: t('staking.stakeTooltip')
+};
+
+const unstakeAction = {
+  label: t('unstake'),
+  loadingLabel: t('staking.unstaking'),
+  confirmingLabel: t('confirming'),
+  action: unstakeBPT,
+  stepTooltip: t('staking.unstakeTooltip')
+};
+
 /**
  * STATE
  */
 const isLoadingApprovalsForGauge = ref(false);
-const isStakingConfirmed = ref(false);
+const isActionConfirmed = ref(false);
 const confirmationReceipt = ref<TransactionReceipt>();
-const bptBalance = ref(balanceFor(props.pool.address).toString());
+const stakeActions = ref<TransactionActionInfo[]>([]);
+const stakedShares = ref('0');
```

Comment:
Receiving `stakedShares` from contract pending response from Fabio regarding whether `balance` from gauge subgraph is cumulative or not.